indico/modules/attachments/operations.py

```diff
@@ -0,0 +1,36 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from flask import session
+
+from indico.core import signals
+from indico.core.db import db
+from indico.modules.attachments import logger
+from indico.modules.attachments.models.attachments import Attachment, AttachmentType
+from indico.modules.attachments.models.folders import AttachmentFolder
+
+
+def add_attachment_link(form, linked_object):
```

Comment:
In the end you are only using it in a single place (at least I couldn't find anything calling this in indico/indico-plugins#28). Anyway, for the sake of consistency with other operations I would rather use pass a `data` dictionary instead of the form and then use `link.populate_from_dict()`. It currently does not support `skip` but that would be easy to adapt (almost copy&paste from `IndicoForm.populate_obj`)